sdks/go/pkg/beam/transforms/periodic/periodic_test.go

```diff
@@ -0,0 +1,57 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package periodic
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/apache/beam/sdks/v2/go/pkg/beam"
+	"github.com/apache/beam/sdks/v2/go/pkg/beam/runners/prism"
+	"github.com/apache/beam/sdks/v2/go/pkg/beam/testing/passert"
+)
+
+func TestSequence(t *testing.T) {
+	p, s := beam.NewPipelineWithRoot()
+	sd := SequenceDefinition{
+		Interval: time.Second,
+		Start:    0,
+		End:      time.Minute.Milliseconds(),
+	}
+	in := beam.Create(s, sd)
+	out := Sequence(s, in)
+	passert.Count(s, out, "SecondsInMinute", 60)
+	beam.Init()
```

Comment:
Two things:
First, `beam.Init()` should be called before pipeline construction, since it's what enables the binary to behave as either a worker or a main. 

Second, calling beam.Init in tests isn't recommended as a rule since it can/will get called multiple times. 

So we have the PTest package to call it after test binary flag parsing, by overriding the default TestMain.

If you add the following:
```
func TestMain(m *testing.M) {
	os.Exit(ptest.MainRetWithDefault(m, "prism"))
}
```

You can even then trigger pipeline execution with `ptest.RunAndValidate(t, p)`, which will fail if an error is returned, but also check any counters from the returned pipeline result, which is another way we could validate this behavior other than `passert.Count`.